year18/src/day11.rs

```diff
@@ -1,10 +1,54 @@
 // Potential improvements:
 //
+use std::i32::MIN;
+use nalgebra::DMatrix;
 
-pub fn day11(_input_lines: &[Vec<String>]) -> (String, String) {
-    let answer1 = 0;
-    let answer2 = 0;
-    (format!("{}", answer1), format!("{}", answer2))
+const POWER_GRID_SIZE: usize = 300;
+
+pub fn day11(input_lines: &[Vec<String>]) -> (String, String) {
+    let serial_number = input_lines[0][0].parse::<i32>().unwrap();
+
+    // The question is 1-indexed, but the matrix is 0-indexed.
+    let power_grid = DMatrix::<i32>::from_fn(
+        POWER_GRID_SIZE, 
+        POWER_GRID_SIZE, 
+        |y, x| ((x as i32 + 11) * (y as i32 + 1) + serial_number) * (x as i32 + 11) / 100 % 10 - 5
+    );
+    
+    let mut max_power: i32 = MIN;
+    let mut max_indices= (0 ,0);
+    let subgrid_size = 3;
+    for jj in 0..=POWER_GRID_SIZE-subgrid_size {
+        for ii in 0..=POWER_GRID_SIZE-subgrid_size {
+            let new_power = power_grid.slice((jj,ii),(subgrid_size,subgrid_size)).sum();
+            if new_power > max_power {
+                max_power = new_power;
+                max_indices = (ii+1, jj+1);
+            }
+        }
+    }
```

Comment:
This for loop is pretty clear, however you can write this in a more functional style as:
```rust
let (ii, jj) = (0..=POWER_GRID_SIZE-subgrid_size).flat_map(|jj| (0..=POWER_GRID_SIZE-subgrid_size).map(|ii| (ii, jj)))
// or if you have itertools: iproduct!(0..=POWER_GRID_SIZE-subgrid_size, 0..=POWER_GRID_SIZE-subgrid_size)
    .max_by_key(|(ii, jj)| power_grid.slice((jj,ii),(subgrid_size,subgrid_size)).sum());
```